src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/PreinitializationManager.cs

```diff
@@ -128,6 +148,14 @@ public void LogStatistics(Logger logger)
             }
         }
 
+        public void CheckPreInitializedAttribute(Logger logger)
```

Comment:
Given the above two comments, I think the scope of the change we need is just:

* Augment LogStatistics to check for the attribute (it will be the only spot that checks for the attribute)
* Make it log the new message even if the logger is not verbose (remove the big `if (logger.IsVerbose)` check and make it local to the existing LogMessage call.

The ConcurrentDictionary, CheckPreInitializedAttribute can go.